packages/jupyter-ai/jupyter_ai/mcp/mcp_config_loader.py

```diff
@@ -0,0 +1,107 @@
+import json
+from pathlib import Path
+from typing import Any
+
+from jsonschema import ValidationError, validate
+from jsonschema.exceptions import SchemaError
+
+
+class MCPConfigLoader:
+    """Loader for MCP server configuration files with JSON schema validation."""
+
+    def __init__(self):
+        # Load the schema from the schema.json file
+        schema_path = Path(__file__).parent / "schema.json"
+        with open(schema_path) as f:
+            self.schema = json.load(f)
+
+        # Cache for storing configurations and their modification times
+        # Key: config_path (str), Value: (config_dict, last_modified_time)
+        self._cache: dict[str, tuple[dict[str, Any], float]] = {}
+
+    def get_config(self, jupyter_dir: str) -> dict[str, Any]:
+        """
+        Read and validate an MCP server configuration file from .jupyter/mcp/config.json.
+
+        Args:
+            jupyter_dir (str): Path to the .jupyter directory
+
+        Returns:
+            Dict[str, Any]: The validated configuration object
+
+        Raises:
+            FileNotFoundError: If the config file doesn't exist
+            json.JSONDecodeError: If the JSON is malformed
+            ValidationError: If the config doesn't match the schema
+            SchemaError: If there's an issue with the schema itself
+        """
+        config_path = Path(jupyter_dir) / "mcp" / "config.json"
+        config_path_str = str(config_path)
+
+        # Check if file exists
+        if not config_path.exists():
+            raise FileNotFoundError(f"Configuration file not found: {config_path}")
+
+        # Get current file modification time
+        current_mtime = config_path.stat().st_mtime
```

Comment:
This can be obtained through `ContentsManager`, but that would make this an async method. Fine to leave it as-is if we are prioritizing for local FS support before remote/network FS support.